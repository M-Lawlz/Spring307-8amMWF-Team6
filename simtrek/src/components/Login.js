import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import {FormControl, FormGroup, FormLabel} from '@material-ui/core';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

class Login extends React.Component {
    constructor(props) {
        super(props);
        this.handleLoginClose = this.handleLoginClose.bind(this);
        this.state = {
            open : (props.loginAttempt) ? true : false
        }
    }

    handleLoginOpen() {
        this.setState({
            open: true
        });
    }

    handleLoginClose() {
        console.log("Handle close in the login component");
        this.setState({
            open: false
        });
        // attempt to hard rerender the parent component but fails
        // this.props.rerenderCallback();

        // pass false (for login popup) back up to the parent
        this.props.loginAttempt(false);
    }

    // TODO: Implement
    submitLoginForm() {}


    // will deal with form stuff later
    render() {
        return (
            <Dialog 
            modal = {true}
            autoDetectWindowHeight={true} 
            autoScrollBodyContent={true}
            contentStyle={{height: "200px", width: "300xp"}}
            onClose={() => this.handleLoginClose()} 
            open={this.state.open}>
                <DialogTitle>Login</DialogTitle>
                <DialogContent>
                <FormGroup controlId="username">
                    {/* <FormLabel>Email or Username</FormLabel>
                    <FormControl autoFocus type="email"/> */}
                    <TextField label="Username" margin="dense"></TextField>
                </FormGroup>
                <FormGroup controlId="password">
                    {/* <FormLabel>Password</FormLabel> */}
                    <TextField label="Password" margin="dense"></TextField>
                <FormControl/>
                </FormGroup>
                </DialogContent>
                <DialogActions>
                    <Button onClick={() => this.handleLoginClose()} color="primary">
                        Cancel
                    </Button>
                    <Button onClick={() => this.submitLoginForm()} color="primary">
                        Login
                    </Button>
                </DialogActions>
            </Dialog>
        );
    }
}

export default Login;
