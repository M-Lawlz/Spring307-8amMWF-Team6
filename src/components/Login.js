import App from "firebase/app";
import "firebase/auth";
import { Button, Dialog, DialogContent, DialogTitle } from "@material-ui/core";
import Form from "react-bootstrap/Form";
import React from "react";

export default class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      isForgotPassword: false,
      open: props.loginShowing ? true : false,
      password: "",
      validated: false,
    };
  }

  attemptLogin = () => {
    App.auth()
      .setPersistence(App.auth.Auth.Persistence.LOCAL)
      .catch(this.handleError);
    App.auth()
      .signInWithEmailAndPassword(this.state.email, this.state.password)
      .then(this.login)
      .catch(this.handleError);
  };

  checkFormValidation(event) {
    event.preventDefault();
    if (event.currentTarget.checkValidity() !== false) {
      this.setState({
        validated: true,
      });
    }
  }

  closeLogin = () => {
    this.setState({
      open: false,
    });
    this.props.loginShowing(false);
  };

  finishResetPassword = () => {
    alert("Password reset email sent!");
    this.toggleForgotPassword();
  }

  handleEmailInput = (email) => {
    this.setState({ email: email.target.value });
  };

  handleError = (error) => {
    alert(error.message);
  };

  handlePasswordInput = (password) => {
    this.setState({ password: password.target.value });
  };

  login = () => {
    alert("Successfully signed in!");
    this.closeLogin();
  };

  sendPasswordResetEmail = (email) => {
    App.auth()
      .sendPasswordResetEmail(email)
      .then(this.finishResetPassword)
      .catch(this.handleError);
  };

  submitForm = (event) => {
    this.checkFormValidation(event);
    if (this.state.isForgotPassword) {
      this.sendPasswordResetEmail(this.state.email);
    } else {
      this.attemptLogin();
    }
  }

  toggleForgotPassword = () => {
    this.setState({
      isForgotPassword: !this.state.isForgotPassword,
    });
  };

  render() {
    if (this.state.isForgotPassword) {
      return (
        <Dialog
          autoDetectWindowHeight={true}
          autoScrollBodyContent={true}
          contentStyle={{ height: "200px", width: "300px" }}
          modal={true}
          onClose={this.toggleForgotPassword}
          open={this.state.open}
        >
          <DialogTitle style={{ textAlign: "center" }}>
            Forgot Password
          </DialogTitle>
          <DialogContent>
            <Form
              onSubmit={this.submitForm}
              validated={this.state.validated}
            >
              <Form.Text>
                If you've forgotten your password, please enter your email
                below. If it exists in our system, you'll receive an email with
                instructions on how to reset your password shortly.
              </Form.Text>
              <br />
              <br />
              <Form.Label style={{ alignSelf: "center" }}>Email</Form.Label>
              <Form.Group controlId={"formEmail"}>
                <Form.Control
                  name={"formEmail"}
                  onInput={this.handleEmailInput}
                  placeholder={"Email"}
                  required
                  style={{ width: "200px" }}
                  type={"email"}
                  value={this.state.email}
                />
              </Form.Group>
              <br />
              <Button color={"primary"} onClick={this.toggleForgotPassword}>
                Cancel
              </Button>
              <Button color={"primary"} type={"Submit"}>
                Send Email
              </Button>
            </Form>
          </DialogContent>
        </Dialog>
      );
    } else {
      return (
        <Dialog
          autoDetectWindowHeight={true}
          autoScrollBodyContent={true}
          contentStyle={{ height: "200px", width: "300px" }}
          modal={true}
          onClose={this.closeLogin}
          open={this.state.open}
        >
          <DialogTitle style={{ textAlign: "center" }}>Log In</DialogTitle>
          <DialogContent>
            <Form
              onSubmit={this.submitForm}
              validated={this.state.validated}
            >
              <Form.Label>Email</Form.Label>
              <Form.Group controlId={"formEmail"}>
                <Form.Control
                  name={"formEmail"}
                  onInput={this.handleEmailInput}
                  placeholder={"Email"}
                  required
                  type={"email"}
                  value={this.state.email}
                />
              </Form.Group>
              <br />
              <Form.Label>Password</Form.Label>
              <Form.Group controlId={"formPassword"}>
                <Form.Control
                  name={"formPass"}
                  onInput={this.handlePasswordInput}
                  placeholder={"Password"}
                  required
                  type={"password"}
                  value={this.state.password}
                />
              </Form.Group>
              <Button
                color={"primary"}
                style={{ fontSize: 10, textAlign: "left" }}
                onClick={this.toggleForgotPassword}
              >
                Forgot Password?
              </Button>
              <br />
              <br />
              <Button color={"primary"} onClick={this.closeLogin}>
                Cancel
              </Button>
              <Button color={"primary"} type={"submit"}>
                Login
              </Button>
            </Form>
          </DialogContent>
        </Dialog>
      );
    }
  }
}
