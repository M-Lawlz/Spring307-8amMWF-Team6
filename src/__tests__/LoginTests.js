import Login from "../components/Login";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

test("Login renders successfully", () => {
  const login = shallow(<Login />);
  expect(toJson(login)).toMatchSnapshot();
});
