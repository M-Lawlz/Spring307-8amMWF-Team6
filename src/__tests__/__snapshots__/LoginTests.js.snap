// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders successfully. 1`] = `
<WithStyles(ForwardRef(Dialog))
  autoDetectWindowHeight={true}
  autoScrollBodyContent={true}
  contentStyle={
    Object {
      "height": "200px",
      "width": "300px",
    }
  }
  modal={true}
  onClose={[Function]}
  open={false}
>
  <WithStyles(ForwardRef(DialogTitle))
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Log In
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <Form
      inline={false}
      onSubmit={[Function]}
      validated={false}
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Email
      </FormLabel>
      <FormGroup
        controlId="formEmail"
      >
        <FormControl
          name="formEmail"
          onInput={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </FormGroup>
      <br />
      <FormLabel
        column={false}
        srOnly={false}
      >
        Password
      </FormLabel>
      <FormGroup
        controlId="formPassword"
      >
        <FormControl
          name="formPass"
          onInput={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </FormGroup>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        style={
          Object {
            "fontSize": 10,
            "textAlign": "left",
          }
        }
      >
        Forgot Password?
      </WithStyles(ForwardRef(Button))>
      <br />
      <br />
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        type="submit"
      >
        Login
      </WithStyles(ForwardRef(Button))>
    </Form>
  </WithStyles(ForwardRef(DialogContent))>
</WithStyles(ForwardRef(Dialog))>
`;
